jwst/assign_wcs/util.py

```diff
@@ -366,15 +366,30 @@ def compute_fiducial(wcslist, bounding_box=None):
     spatial_footprint = footprints[spatial_axes]
     spectral_footprint = footprints[spectral_axes]
 
+    wcs_init = wcslist[0]
+    domain_bounds = np.hstack([wcs_init.backward_transform(
+        *w.footprint(bounding_box=bounding_box).T) for w in wcslist])
+
+    output_bounding_box = []
+    for axis in wcslist[0].output_frame.axes_order:
+        axis_min, axis_max = domain_bounds[axis].min(), domain_bounds[axis].max()
+        output_bounding_box.append((axis_min, axis_max))
+
+    sky_bounds = np.array(wcs_init.forward_transform(*output_bounding_box))
+    fiducial_new = np.mean(sky_bounds, axis=1)
+    return fiducial_new
+
     fiducial = np.empty(len(axes_types))
     if spatial_footprint.any():
         lon, lat = spatial_footprint
         lon, lat = np.deg2rad(lon), np.deg2rad(lat)
-        x_mean = np.mean(np.cos(lat) * np.cos(lon))
-        y_mean = np.mean(np.cos(lat) * np.sin(lon))
-        z_mean = np.mean(np.sin(lat))
-        lon_fiducial = np.rad2deg(np.arctan2(y_mean, x_mean)) % 360.0
-        lat_fiducial = np.rad2deg(np.arctan2(z_mean, np.sqrt(x_mean ** 2 + y_mean ** 2)))
+        x_mid = (np.max(np.cos(lat) * np.cos(lon)) +
+                 np.min(np.cos(lat) * np.cos(lon))) / 2.
+        y_mid = (np.max(np.cos(lat) * np.sin(lon)) +
+                 np.min(np.cos(lat) * np.sin(lon))) / 2.
+        z_mid = (np.max(np.sin(lat)) + np.min(np.sin(lat))) / 2.
+        lon_fiducial = np.rad2deg(np.arctan2(y_mid, x_mid)) % 360.0
+        lat_fiducial = np.rad2deg(np.arctan2(z_mid, np.sqrt(x_mid ** 2 + y_mid ** 2)))
         fiducial[spatial_axes] = lon_fiducial, lat_fiducial
     if spectral_footprint.any():
         fiducial[spectral_axes] = spectral_footprint.min()
```

Comment:
These lines in the function should be removed entirely as they are completely replaced by the code you added above that works perfectly well based on my own tests with the changes.

Reply:
Yeah, sorry - that commit was a code-share more than a well-written commit with intent to merge 😬 

I'm still seeing a slight exclusion of data in my test - not sure if I'm missing some slight shift somewhere. Also thinking I might need to take the mean in spherical coordinate space to avoid issues crossing 359->0 in RA deg space, but I'll try to clean this up and get a better commit in place.

Reply:
No need to worry.  This is one advantage of the PR, it can be modified as many times as needed before being merged, and allows (as you are doing here) others to see what you are doing.  I know I have thought I removed such extra code before only to find it there later.  So far, so good.

Reply:
I think it's better to work in 3D cartesian space than in spherical coordinates. And yes, there's still an ofset.